homeassistant/components/switch/flux.py

```diff
@@ -174,14 +199,22 @@ def flux_update(self, now=dt_now()):
                 temp = self._sunset_colortemp - temp_offset
             else:
                 temp = self._sunset_colortemp + temp_offset
-            x_val, y_val, b_val = color_RGB_to_xy(*temp_to_rgb(temp))
-            brightness = self._brightness if self._brightness else b_val
-            set_lights_xy(self.hass, self._lights, x_val,
-                          y_val, brightness)
-            _LOGGER.info("Lights updated to x:%s y:%s brightness:%s, %s%%"
-                         " of night cycle complete at %s", x_val, y_val,
-                         brightness, round(percentage_of_night_complete*100),
-                         as_local(now))
+            if self._mode == 'xy':
```

Comment:
This entire block is duplicate of lines 170-185, pull these out into a separate method